File: Compare within margin.py
def close_compare(*args):
    a,b,margin = args[0],args[1],0
    if len(args)==2:
        return -1 if a < b else 1 if a > b else 0
    else:
        margin = args[2]
        dis = abs(a-b)
        if dis <= margin:
            return 0
        else:
            return -1 if a < b else 1 if a > b else 0
        

File: Compare within margin/Compare within margin.js
function closeCompare(...args){
  const [a,b,...margin] = args
  if(args.length==2){
    if(a<b){
      return -1
    }else if(a>b){
      return 1
    }else{
      return 0
    }
  }else{
    let dis = Math.abs(a-b)
    if(dis <= margin){
      return 0
    }else{
      if(a<b){
        return -1
      }else if(a>b){
        return 1
      }else{
        return 0
      }
    }
  }
}

